### PR TITLE
added naiveBayes e1071

### PR DESCRIPTION
**Pull request by erblast**
added naiveBayes e1071

### CODE REVIEW

**Comment by erblast** [comment]
hi i am fairly new to this github thing, i hope i have figured it out now how to do proper pull requests.

File: R/twidlr-e1071.R [hunk 1]
```diff
@@ -33,3 +33,48 @@ predict.svm <- function(object, data, ...) {
...
+  UseMethod("naiveBayes")
+}
+
+#' @export
+naiveBayes.default <- function(formula, data, ...) {
+  naiveBayes.data.frame(as.data.frame(data), formula, ...)
+}
+
+#' @export
+naiveBayes.data.frame <- function(data, formula, ...) {
```
Comment (drsimonj) [inline R/twidlr-e1071.R]: Please remove this `.data.frame` method and place code in `.default` method

File: R/twidlr-e1071.R [hunk 2]
```diff
@@ -33,3 +33,48 @@ predict.svm <- function(object, data, ...) {
...
+#' # Help page for function being twiddled
+#' ?e1071::naiveBayes
+#'
+naiveBayes <- function(data, formula, ...) {
+  check_pkg("e1071")
+  UseMethod("naiveBayes")
+}
+
+#' @export
+naiveBayes.default <- function(formula, data, ...) {
```
Comment (drsimonj) [inline R/twidlr-e1071.R]: Please follow new convention of using `coerce_args()` to coerce `data` and `formula` appropriately. For example, see https://github.com/drsimonj/twidlr/blob/master/R/twidlr-e1071.R#L21-L27

**Comment by drsimonj** [comment]
Looking good, and thanks for wanting to contribute to twidlr! I've left few comments on your changes. Please make required changes when possible and let me know.

File: R/twidlr-e1071.R [hunk 3]
```diff
@@ -33,3 +33,48 @@ predict.svm <- function(object, data, ...) {
...
+  naiveBayes.data.frame(as.data.frame(data), formula, ...)
+}
+
+#' @export
+naiveBayes.data.frame <- function(data, formula, ...) {
+  e1071::naiveBayes(formula = formula, data = data, ...)
```
Comment (drsimonj) [inline R/twidlr-e1071.R]: If available, please use `e1071:::naiveBayes.formula`

**Comment by erblast** [comment]
So i think that should do it. If that is alright I would also like to add `gamlss()` next which I frequently use. Many thanks for coming up with these great tools.  Just  discovered `pipelearner` the other week and started writing wrapper functions for all the models I wanted to use. Then I saw you blogpost about twidlr and thought this is much better than anything i would have come up with

**Comment by drsimonj** [comment]
That's done it. Thanks again for a great contribution! And of course, please add any other models 👍 

And so glad to hear you're trying out pipelearner. As you've done, I started finding bottlenecks for pipelearner like having to wrap model functions. So, hopefully, twidlr is the first in a family of more targetted and modular packages that will work together to do tidy modelling in R. Please feel free to send any ideas as you go.

**Pull request merged**